python/ray/serve/_private/client.py

```diff
@@ -107,54 +106,12 @@ def shutdown(self) -> None:
             del self.handle_cache[k]
 
         if ray.is_initialized() and not self._shutdown:
-            ray.get(self._controller.shutdown.remote())
-            self._wait_for_deployments_shutdown()
-
-            ray.kill(self._controller, no_restart=True)
-
-            # Wait for the named actor entry gets removed as well.
-            started = time.time()
-            while True:
-                try:
-                    ray.get_actor(self._controller_name, namespace=SERVE_NAMESPACE)
-                    if time.time() - started > 5:
-                        logger.warning(
-                            "Waited 5s for Serve to shutdown gracefully but "
-                            "the controller is still not cleaned up. "
-                            "You can ignore this warning if you are shutting "
-                            "down the Ray cluster."
-                        )
-                        break
-                except ValueError:  # actor name is removed
-                    break
-
+            try:
+                ray.get(self._controller.graceful_shutdown.remote())
+            except ray.exceptions.RayActorError:
+                logger.info("The controller has already shut down.")
```

Comment:
This is worded a bit confusingly for a user-facing message. I would probably omit logging at all here and instead have a log statement from the controller when shutdown completes.